pkg/applier/applier_test.go

```diff
@@ -374,8 +376,7 @@ func TestApply(t *testing.T) {
 				Mapper:     fakeClient.RESTMapper(),
 				// TODO: Add tests to cover status mode
 			}
-			applier, err := NewNamespaceSupervisor(cs, syncScope, syncName, 5*time.Minute)
-			require.NoError(t, err)
+			applier := NewSupervisor(cs, syncScope, syncName, 5*time.Minute)
```

Comment:
do we want to call AnnotateStatusMode in these tests?

Reply:
If not, we need a test for that method explicitly

Reply:
Added a separate unit test